File: CompSci 1/Unit 14, Iteration - while/GUI08/GUI08.java
import java.awt.*;
import javax.swing.*;
import java.awt.event.*;

// Step 1: make frame a listener
public class GUI08 extends JFrame implements ActionListener {

    // instance variables
    private JButton button;
    private JButton circle;

    // constructor
    public GUI08()
    {
        // create button
        button = new JButton("Change Color");
        circle = new JButton("Draw Circle");

        // set button attributes
        button.setLocation(175, 50);
        button.setSize(150, 50);

        circle.setLocation(175,125);
        circle.setSize(150, 50);

        // add button to frame
        getContentPane().add(button);
        getContentPane().add(circle);

        // Step 2: register listener with button
        button.addActionListener(this);
        circle.addActionListener(this);

        // set frame attributes
        setLayout(null);
        setSize(500, 500);
        setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
        setVisible(true);
    }

    // Step 3: add an actionPerformed method
    public void actionPerformed(ActionEvent event)
    {
        if(event.getSource() == button)             // identify source
        {
            int r = (int)(Math.random() * 256);     // red intensity: 0..255
            int g = (int)(Math.random() * 256);     // green intensity: 0..255
            int b = (int)(Math.random() * 256);     // blue intensity:0..255
            Color color = new Color(r, g, b);       // make custom color
            getContentPane().setBackground(color);  // change background of frame
        }

        if(event.getSource() == circle)             // identify source
        {
            Graphics page = getGraphics();
            int r = (int)(Math.random() * 256);     // red intensity: 0..255
            int g = (int)(Math.random() * 256);     // green intensity: 0..255
            int b = (int)(Math.random() * 256);     // blue intensity:0..255
            int x = (int)(Math.random() * 500);
            int y = (int)(Math.random() * 500);
            page.setColor(new Color(r, g, b));
            page.fillOval(x, y, 100, 100);
        }
    }

    // main method
    public static void main(String[] args)
    {
        GUI08 app = new GUI08();  // run program
    }
}

File: CompSci 1/Unit 14, Iteration - while/GUI08/GUI08.kt
import java.util.*
import java.awt.*
import javax.swing.*
import java.awt.event.*

// Step 1: make frame a listener
class GUI08 : JFrame(), ActionListener {

    // instance variables
    private val button: JButton
    private val circle: JButton

    // constructor
    init {
        // create button
        button = JButton("Change Color")
        circle = JButton("Draw Circle")

        // set button attributes
        button.setLocation(175, 50)
        button.setSize(150, 50)

        circle.setLocation(175, 125)
        circle.setSize(150, 50)

        // add button to frame
        contentPane.add(button)
        contentPane.add(circle)

        // Step 2: register listener with button
        button.addActionListener(this)
        circle.addActionListener(this)

        // set frame attributes
        layout = null
        setSize(500, 500)
        defaultCloseOperation = JFrame.EXIT_ON_CLOSE
        isVisible = true
    }

    // Step 3: add an actionPerformed method
    override fun actionPerformed(event: ActionEvent) {
        if (event.source === button)
        // identify source
        {
            val r = (Math.random() * 256).toInt()     // red intensity: 0..255
            val g = (Math.random() * 256).toInt()     // green intensity: 0..255
            val b = (Math.random() * 256).toInt()     // blue intensity:0..255
            val color = Color(r, g, b)       // make custom color
            contentPane.background = color  // change background of frame
        }

        if (event.source === circle)
        // identify source
        {
            val page = graphics
            val r = (Math.random() * 256).toInt()     // red intensity: 0..255
            val g = (Math.random() * 256).toInt()     // green intensity: 0..255
            val b = (Math.random() * 256).toInt()     // blue intensity:0..255
            val x = (Math.random() * 500).toInt()
            val y = (Math.random() * 500).toInt()
            page.color = Color(r, g, b)
            page.fillOval(x, y, 100, 100)
        }
    }
}

fun main(args: Array<String>) {
    GUI08()
}